src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts

```diff
@@ -322,61 +324,67 @@ export class UserDataProfileImportExportService extends Disposable implements IU
+				const id = await this.pickProfileContentHandler(profile.name);
+				if (!id) {
+					return;
+				}
+				const profileContentHandler = this.profileContentHandlers.get(id);
+				if (!profileContentHandler) {
+					return;
+				}
+				const saveResult = await profileContentHandler.saveProfile(profile.name, JSON.stringify(profile), CancellationToken.None);
+				if (!saveResult) {
+					return;
+				}
+				const message = localize('export success', "Profile '{0}' is exported successfully.", profile.name);
+				if (profileContentHandler.extensionId) {
 					const result = await this.dialogService.show(
 						Severity.Info,
-						localize('export success', "Profile '{0}' is exported successfully.", profile.name),
-						buttons,
+						message,
+						[localize('copy', "Copy Link"), localize('open', "Open in {0}", profileContentHandler.name), localize('close', "Close")],
 						{ cancelId: 2 }
 					);
 					switch (result.choice) {
 						case 0:
 							await this.clipboardService.writeText(
 								URI.from({
 									scheme: this.productService.urlProtocol,
-									authority: `${UserDataProfileImportExportService.PROFILE_URL_AUTHORITY_PREFIX}${saveResult.id}`,
-									path: `/${saveResult.resource.toString()}`
+									authority: `${UserDataProfileImportExportService.PROFILE_URL_AUTHORITY}`,
+									path: `/${id}/${saveResult.id}`
 								}).toString());
 							break;
 						case 1:
-							await this.openerService.open(saveResult.resource.toString());
+							await this.openerService.open(saveResult.link.toString());
 							break;
 
 					}
+				} else {
+					await this.dialogService.show(Severity.Info, message);
 				}
 			});
 		} finally {
 			disposables.dispose();
 		}
 	}
 
-	private async saveProfileContent(name: string, content: string): Promise<{ resource: URI; id: string } | null> {
-		const id = await this.pickProfileContentHandler(name);
-		if (!id) {
-			return null;
-		}
-		const profileContentHandler = this.profileContentHandlers.get(id);
-		if (!profileContentHandler) {
-			return null;
-		}
-		const resource = await profileContentHandler.saveProfile(name, content, CancellationToken.None);
-		return resource ? { resource, id } : null;
-	}
-
 	private async resolveProfileContent(resource: URI): Promise<string | null> {
 		if (await this.fileService.canHandleResource(resource)) {
 			return this.fileUserDataProfileContentHandler.readProfile(resource, CancellationToken.None);
 		}
 
 		if (this.isProfileURL(resource)) {
-			const handlerId = resource.authority.substring(UserDataProfileImportExportService.PROFILE_URL_AUTHORITY_PREFIX.length);
+			let handlerId: string, idOrUri: string | URI;
+			if (resource.authority === UserDataProfileImportExportService.PROFILE_URL_AUTHORITY) {
+				idOrUri = this.uriIdentityService.extUri.basename(resource);
+				handlerId = this.uriIdentityService.extUri.basename(this.uriIdentityService.extUri.dirname(resource));
```

Comment:
is `basename` redundant here?

Reply:
Nopes, because `dirname` returns the parent URI not the parent name.